### DISCUSSION TITLE
CB1_Debian11_bullseye_minimal Image discussion

### DISCUSSION THREAD

**Discussion by bigtreetech** [bigtreetech/CB1#9] [Ideas]
# CB1_Debian11_bullseye_minimal_kernel4.9_20220730.img
* This image is  minimal Debian11 file system. Please contact [here](https://github.com/bigtreetech/CB1/discussions) for detailed information and feedback

<hr /><em>This discussion was created from the release <a href='https://github.com/bigtreetech/CB1/releases/tag/'>V1.1.0</a>.</em>

# Know issues
* After successfully connecting to WIFI, the system will save the current connection information and automatically connect this WIFI after the next restart. When setting a new WiFi in the configuration file again, the system will first try to connect to the previously saved WiFi after startup. When the previous WiFi cannot be connected, it will try to connect to the newly set WIFI (This connection logic will be optimized in the next version, and the WIFI priority set in the configuration file will be set to the highest)
* The HDMI resolution of this OS does not support non-standard, such as 1024x600, 800x480. We are adapting to the next version and will add this support

# Points to be discussed
1.  The shell tool of the current Image is Debian's own tool, and some command cannot be completed automatically by `Table`. Is it necessary to build a more powerful shell tool (for example: [Z shell](https://github.com/zsh-users/zsh))? Or do you have any recommendations?

# Tested feature
* I installed and tested `Klipper`, `Moonraker`, `Fluidd`, `Mainsail`, and `Octoprint + Klipper` in localhost:5000 port with [Kiauh](https://github.com/th33xitus/kiauh). And All they can be normally connected to the motherboard to control the printer.
* Install the accelerometers according to [Klipper document](https://www.klipper3d.org/Measuring_Resonances.html)
  `sudo apt update`
  `sudo apt install python3-numpy python3-matplotlib libatlas-base-dev`
  `~/klippy-env/bin/pip install -v numpy`
  the libraries can be installed successfully and `TEST_RESONANCES` run normally.

**Comment by frutabruta** [comment]
This seems much better. Some errors still show up after boot, and I guess it has something to do with drivers/device tree.
I would appreciate a way to change resolution: the terminal shows in what seems to be 800x600 on my FullHD HDMI display.
It would be great to find a way to install WaveShare CAN HAT,  so I can use EBB42 with Manta.

**Reply by djscrubb** [reply]
Enabling spi breaks image,
Installed raspi-config
Enabled spi
Reboot
No longer loads
Very very frustrating

**Reply by dahliamoth** [reply]
im assuming you are attempting to setup a adxl for input shaping? I may be at the same hang up as you.

**Reply by emesix** [reply]
Well as i see it, No SPI support by default is no input shaping.

But mine CB1 Debian image is now instable. So maybe test it with a RPI CM4 when they are available....
Would have been nice if it was a little more mature on release.

**Reply by dahliamoth** [reply]
in this upload they claim they tested it and it works. just wish they would tell us how. what klippy is hanging up on on my end is connecting to the secondary mcu that was setup per klipper's doc. refuses to boot if the cfg is calling for that mcu

**Reply by shawnchr** [reply]
Yeah same here with the loading issue. 
Man I love the idea of this but wow this is still really buggy and really unusable/

**Comment by djscrubb** [comment]
Thank you for this, the update is nice

**Comment by Aleks-MO** [comment]
There is no Wi-Fi connection..... I can't configure it. (((((

**Reply by djscrubb** [reply]
you edit system.cfg with wifi settings

**Reply by Aleks-MO** [reply]
https://github.com/bigtreetech/CB1/issues/12

**Comment by shawnchr** [comment]
Ok so I am having an issue My SSID on my wireless has a space in it and I have tried modify the /mnt/udisk/system.cfg with an escape character  WIFI_SSID="xx\ xxxxxxx" but I still getting Error: invalid extra argument 'Wireless'. Any help would be appreciated this work on the rasberry pis no issue even with no escape character.

**Reply by djscrubb** [reply]
How about adding a guest network with no space?

**Reply by shawnchr** [reply]
I could do that but why everything else on my network works fine this way. There is an issue and it needs to be addressed, you do not replace an engine for a flat tire. Plus with a guest network I have to connect to it to use my printers not a great solution.

**Reply by bigtreetech** [reply]
This problem is confirmed, and a new version will be fixed and released soon. Sorry for the inconvenience

**Reply by djscrubb** [reply]
And everybody that bought v1. 0 are just beta testers and stuck with product that are no just stuck with product that just has flaws?

**Reply by shawnchr** [reply]
Well If there is a hardware issue I would hope they would standby it and replace it like they did there skr boards that were broken. I have 3 of these and would really disappointed.

**Reply by djscrubb** [reply]
I agree it'd very disappointing that it's reported that everything installs, how ever there is a difference between installed and actually working ie klippersceeen

**Comment by jceller** [comment]
I can't get my wifi to work either. I have edited the system.cfg file. It appears that mine keeps going up and down.
![wifi](https://user-images.githubusercontent.com/105725351/184559366-6b7c1516-6cdb-4f05-83a8-6ad27d26a713.png)

**Reply by shawnchr** [reply]
I can tell You I got one that is working fine and another one I got to connect for about 2 hours and now can't get it to connect again. I will be emailing support,

**Reply by frutabruta** [reply]
> I can't get my wifi to work either. I have edited the system.cfg file. It appears that mine keeps going up and down. ![wifi](https://user-images.githubusercontent.com/105725351/184559366-6b7c1516-6cdb-4f05-83a8-6ad27d26a713.png)

A had the same problem with both of my CB1 and I am pretty sure it is a SW problem. When I was trying ifconfig wlan0 kept vanishing and reappearing. I flashed back an older Ubuntu based image and the Wi-Fi is stable again.

**Reply by frutabruta** [reply]
> I can tell You I got one that is working fine and another one I got to connect for about 2 hours and now can't get it to connect again. I will be emailing support,

I might be a bad solder joint on Wi-Fi module. I ha to resolder one of mine. The joint probably cracked when I installed the official heatsink.
![IMG_20220815_094804](https://user-images.githubusercontent.com/62122818/184597294-7fee200c-679a-4c14-90f1-dfece35b4549.jpg)

**Reply by shawnchr** [reply]
The kernel crashes too I think it might be multiple issues with this one

**Comment by jceller** [comment]
I tried to email support but it requires you to use an news group server. I
had always heard BTT had such good products, this is the first of them that
I have bought and I am not impressed. This will probably be the last I buy
from them. I tried to open a ticket but the form requires an image and when
I try to add the image it errors out even though it is under their 5MB
limit. It won't let you submit the ticket without the image.

**Reply by shawnchr** [reply]
Not sure why you say it needs news group server I sent an email to the following address service001@biqu3d.com it has not been rejected or returned. I have not received a response but I also included this address info@biqu.equipment which I have gotten responses from.

**Reply by frutabruta** [reply]
> I tried to email support but it requires you to use an news group server. I had always heard BTT had such good products, this is the first of them that I have bought and I am not impressed. This will probably be the last I buy from them. I tried to open a ticket but the form requires an image and when I try to add the image it errors out even though it is under their 5MB limit. It won't let you submit the ticket without the image.
> […](#)
> On Sun, Aug 14, 2022 at 8:02 PM shawnchr ***@***.***> wrote: I can tell You I got one that is working fine and another one I got to connect for about 2 hours and now can't get it to connect again. I will be emailing support, — Reply to this email directly, view it on GitHub <[#9 (reply in thread)](https://github.com/bigtreetech/CB1/discussions/9#discussioncomment-3394913)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AZGT3J5A4NSCLZHQ4YEX6QLVZGJJHANCNFSM55CQ7LXQ> . You are receiving this because you commented.Message ID: ***@***.***>

I can confirm the ticketing form on biqu.equipment does not work, which made me very sad, too. I told them on messenger, but no-one answered. It seems like another do-your-own-support product.

**Comment by jceller** [comment]
If you go to their website and click on the address it asks for a news
group server.

**Comment by jceller** [comment]
Here is what you get if you go to their website and click on the email
address. It won't send unless you have filled in the newsgroup server.

**Comment by jceller** [comment]
That is the same thing mine is doing. The older image doesn't seem to work
anymore due to it appears that ubuntu doesn't have any releases for it. I
tried it before the Bullseye and it failed every time I tried to load it.

**Reply by frutabruta** [reply]
You can upgrade the Ubuntu so that the apt would work again: 
Just type "impish" instead of "cosmic" and "jammy" instead of "bionic" in this tutorial:
https://linuxconfig.org/how-to-downgrade-ubuntu-linux-system-to-its-previous-version
(Yes, despite the name, it works for upgrading)

**Comment by Magic2991** [comment]
Das ist mein erstes BTT Produkte. leider funktionert nicht mal, das auf spielen der image datei auf die Sd karte. egal ob mit meine Config für wlan,nutzer eingaben.
egal wo mit erstellt, etcher, rasyberry pi imager nur fehler meldundungen. die partion mit der config erkennt windows nicht.

**Reply by RcTomcat1** [reply]
Haben Sie der Partition auch einen Laufwerksbuchstaben zugewiesen? Die Partition mit der WIFI Config Datei ist FAT32 formatiert, das kann jedes Windows lesen und schreiben

**Comment by jceller** [comment]
are you using the new image based on debian bullseye?

**Comment by shogowarrior** [comment]
I have spotty Wi-Fi but my main problem is: I don’t see any uart device for the mcu. Is there a way to configure uart for this build?

**Comment by shogowarrior** [comment]
How do I configure UART for this board?

**Reply by shogowarrior** [reply]
@bigtreetech : Can you please tell us how to setup UART on this one? Most unconventional unix build I have seen till date

**Comment by dkt0591** [comment]
i can't make camera via CSI work :(

**Reply by emesix** [reply]
![image](https://user-images.githubusercontent.com/1113726/185868798-75aa8ef6-585b-4fc2-96da-96fe6039e25b.png)
I have the same problem, but the cause might be different.

**Reply by dkt0591** [reply]
> ![image](https://user-images.githubusercontent.com/1113726/185868798-75aa8ef6-585b-4fc2-96da-96fe6039e25b.png) I have the same problem, but the cause might be different.
i read [frutabruta](https://github.com/frutabruta) say CB1 not support CSI, DSI port @.@,  this is a big disappointment :(

**Reply by pawelksi** [reply]
USB cameras work very well. I have AUSDOM AF640 and it works fine. Personally, I recommend USB cameras, connecting the camera to the CSI with a cable longer than 20cm may cause a problem. The camera on the USB cable is much more convenient and universal.

**Reply by dkt0591** [reply]
> AUSDOM AF640

Thank you for advise @pawelksi, because i already have a Pi cam (clone) so i'm looking CSI to USB adapter but found nothing (have some CSI-2 to USB at aliexpress but i dont think it work for my Pi Cam because its CSI not CSI-2 :( ) , sigh. I think the one you mention above or Wyze Cam v3 (with modify focus screw for manual focus) will work okay, start buying them

**Reply by pawelksi** [reply]
I had a lot of problems with the Pi Cam and the genuine RPi as I connected the 50cm FCC cable. The picture was unstable and sometimes it was not there at all. I bought the Ausdom camera for $ 15 on Aliexpress, it was a promotion.

**Comment by dkt0591** [comment]
And i got random freeze system in middle of print also, sigh, i tried to replaced usb cable, tried other mirco sd card, buy a new official Raspberry Power Supply 5.1V 3A also, still random freeze and crash middle print, i dont know if cause OS system unstable or hardware problem, sighhhhhhhhhhhhhhh
But i think problem not because data usb cable, because when printer freeze, CB1 freeze too, i can't connect to it, but led lights still on

**Reply by basha318i** [reply]
> And i got random freeze system in middle of print also, sigh, i tried to replaced usb cable, tried other mirco sd card, buy a new official Raspberry Power Supply 5.1V 3A also, still random freeze and crash middle print, i dont know if cause OS system unstable or hardware problem, sighhhhhhhhhhhhhhh
> But i think problem not because data usb cable, because when printer freeze, CB1 freeze too, i can't connect to it, but led lights still on

you might be getting kernel panics; try to go back to Hurakan v 1.0.1; seems to be the most stable for now.

**Reply by dkt0591** [reply]
> > And i got random freeze system in middle of print also, sigh, i tried to replaced usb cable, tried other mirco sd card, buy a new official Raspberry Power Supply 5.1V 3A also, still random freeze and crash middle print, i dont know if cause OS system unstable or hardware problem, sighhhhhhhhhhhhhhh
> > But i think problem not because data usb cable, because when printer freeze, CB1 freeze too, i can't connect to it, but led lights still on
> 
> you might be getting kernel panics; try to go back to Hurakan v 1.0.1; seems to be the most stable for now.

Thank you basha318i, so far it is, I tried 1.0.2.1 (Hurakan), after 24h wifi still connected (with Debian Bullseye i will disconnect after 6 hours and random disconnect also ) stable and didn't freeze after 10 prints. Thank you so much ;)

**Reply by pawelksi** [reply]
I also have Debian crashing. An error is displayed in SSH.
Message from syslogd@localhost at Aug 25 18:37:07 ...kernel:Process ksdioirqd/mmc1 (pid: 25628, stack li mit = 0xffffffc03d940000)

I bought the CB1 with the M8P for the printer, not to play the beta tester.
I have installed Ubuntu which is not supported by BTT anymore and so far it works fine.

**Reply by dkt0591** [reply]
@pawelksi: yes i installed 1.0.2.1 (Hurakan) for likely 8 days, very stable without any errors or crashes, but have to disable hibernate mode and set "wifi.powersave=2" (disable power saving) for wifi to keep it online day after day

**Reply by pawelksi** [reply]
Well, I was just wondering why after a long work I have to restart the system, and it just hibernated ..

**Comment by ghent360** [comment]
CB1 with bullseye image just freezes after 5 minutes. HDMI output is distorted.

**Reply by RcTomcat1** [reply]
Mine is also distorted but "clears up" after a few minutes.
My system does not freeze but I find it laggy as hell.
The ubuntu version was definitely smoother

**Comment by Fives3Dprint** [comment]
Hi! I installed octoprint on cb1, works fine but the wifi isn't available, I only can use it by cable, I use the last debian image

**Reply by Fives3Dprint** [reply]
And I can't remove the plastic in

**Reply by RcTomcat1** [reply]
> And I can't remove the plastic in

The 3M protective film is split in two pieces in my case: both parts are easy to peel of.

**Comment by stas2z** [comment]
just gona leave it here
https://github.com/stas2z/orangepi-build-cb1/releases/
;)

p.s.
probably guys from #7 will be interested also
@pendor

**Reply by stas2z** [reply]
Adxl host connection with this image

enable spi overlay
![Selection_910](https://user-images.githubusercontent.com/5320135/188259292-19f4c7fa-d501-4052-a13c-b01ad739a5ae.jpg)

setup adxl in the config
![Selection_911](https://user-images.githubusercontent.com/5320135/188259288-7c4027a7-3d5a-44a1-994a-c0e8da48f63f.jpg)
cb1 dont have both spi1 CS pins on 40pin connector, so im using PC10 pin as software CS 
and voi la :)

![Selection_912](https://user-images.githubusercontent.com/5320135/188259343-0007acd7-db20-4404-a6ca-215eae995cc6.jpg)

actual wiring

![photo_2022-09-03_09-57-10](https://user-images.githubusercontent.com/5320135/188259719-8b42b691-c7c4-47e2-b47a-1b07f7475e37.jpg)
probably the same can be made with btt images, at least with hurakan0711

**Reply by issey83** [reply]
thank you, i got my cb1+ pi4b adapter + adxl345 working, with the image you shared.

**Reply by L0615T1C5-216AC-9437** [reply]
My manta m8p+cb1 works thanks to this! Thank you a lot

**Reply by mutlubasdas** [reply]
> thank you, i got my cb1+ pi4b adapter + adxl345 working, with the image you shared.

For the pi4b adapter, did you use this wiring? And it works like this?

**Reply by acwalter** [reply]
Yes for me with cb1+pi4b and OrangePi image

**Reply by DrakeTech-hub** [reply]
> My manta m8p+cb1 works thanks to this! Thank you a lot

can you kindly tell me the pin names used on the m8p? Thank you

**Comment by shawnchr** [comment]
Thx

**Comment by jceller** [comment]
For the CB1 it is biqu/biqu

**Reply by shawnchr** [reply]
not for this image its not

**Reply by acwalter** [reply]
orangepi/orangepi

**Comment by jceller** [comment]
I just heard from their support and they are telling me to change to a
different wifi network. They clearly have no intention of standing behind
their product.

**Comment by jceller** [comment]
That is not the type of company I want to spend my money with. I heard from
their support and they just sent me the install doc and told me to do 5.4
of it which
I had already done.

**Comment by jceller** [comment]
I have just tried to send one to the emails you listed and will wait to see
their response.

**Comment by jceller** [comment]
I will look and see if that might be the case. Thanks for the info.

**Comment by airbornetrooper82573** [comment]
Is it possible to change the boot image or add in a splash video? I have the CB1 connected to BTT Manta M8P and using the BTT HDMI5 screen.
